src/test-helpers/typeFactory.ts

```diff
@@ -0,0 +1,77 @@
+import { ApiPromise, WsProvider } from '@polkadot/api';
+import { Metadata } from '@polkadot/metadata';
+import { Option, TypeRegistry } from '@polkadot/types';
+import { StorageKey, Tuple, Vec } from '@polkadot/types';
+import { ParaId } from '@polkadot/types/interfaces';
+import {
+	Codec,
+	Constructor,
+	InterfaceTypes,
+	Registry,
+} from '@polkadot/types/types';
+
+import { rococoMetadataV228 } from './metadata/rococoMetadata';
+
+export function createApiWithAugmentations(): ApiPromise {
+	const registry = new TypeRegistry();
+	const metadata = new Metadata(registry, rococoMetadataV228);
+
+	registry.setMetadata(metadata);
+
+	const api = new ApiPromise({
+		provider: new WsProvider('ws://', false),
+		registry,
+	});
+
+	api.injectMetadata(metadata, true);
+
+	return api;
+}
+
+export class TypeFactory {
+	readonly #api: ApiPromise;
+	readonly #registry: Registry;
+
+	constructor(api: ApiPromise) {
+		this.#api = api;
+		this.#registry = this.#api.registry;
+	}
+
+	public storageKey = (index: number): StorageKey => {
+		const key = new StorageKey(
+			this.#registry,
+			this.#api.query.crowdloan.funds.key(this.paraIndex(index))
+		);
+
+		return key.setMeta(this.#api.query.crowdloan.funds.creator.meta);
```

Comment:
What is `creator`?

Reply:
`creator` allows you to access the metadata to decode the necessary args as well as passing necessary type information to unwrap the `StorageType` and set it to the output Type.